ra/ra.go

```diff
@@ -897,16 +897,35 @@ func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, authzs []*c
 			ch <- err
 		}(authz)
 	}
-	var caaFailures []string
-	for _ = range authzs {
-		if err := <-ch; berrors.Is(err, berrors.CAA) {
-			caaFailures = append(caaFailures, err.Error())
-		} else if err != nil {
-			return err
+	var subErrors []berrors.SubBoulderError
+	var firstBadIdent *identifier.ACMEIdentifier
+	for _, authz := range authzs {
+		err := <-ch
+		if err != nil {
+			if firstBadIdent == nil {
+				firstBadIdent = &authz.Identifier
+			}
+			if bErr, ok := err.(*berrors.BoulderError); ok && berrors.Is(bErr, berrors.CAA) {
```

Comment:
`errors.Is` does the casting check itself, no need to do it here.

Reply:
Good call, fixed in 6139b8b